modules/geo/src/main/java/org/opensearch/geo/search/aggregations/bucket/geogrid/InternalGeoGrid.java

```diff
@@ -54,7 +54,7 @@
  * All geo-grid hash-encoding in a grid are of the same precision and held internally as a single long
  * for efficiency's sake.
  *
- * @opensearch.internal
+ * @opensearch.api
  */
 public abstract class InternalGeoGrid<B extends InternalGeoGridBucket> extends InternalMultiBucketAggregation<
```

Comment:
Should we rename these classes to get rid if `InternalXxx` prefix? It does not look like API for public use, naming should be appropriate

Reply:
Alternatively, we could introduce interfaces (as public APIs) and keep these classes indeed internal

Reply:
We already have public GeoGrid/GeoGridBucket interfaces. I can rename from InternalGeoGrid to AbstractGeoGrid. Do you have any other suggestions?

Reply:
May be `GeoGridBase` ? The classes are abstract already ....

Reply:
Only in terms of aggregation:
In my mind Internal prefix was added to classes which will be used when data nodes send the response to coordinator for aggregation. Which made sense to me. 

Then the other classes GeoGridBucket classes are the actual interfaces which provides the response to customer.

Reply:
> Only in terms of aggregation: In my mind Internal prefix was added to classes which will be used when data nodes send the response to coordinator for aggregation. Which made sense to me.

But they are also abstract, which means they only serve as building blocks, could not be instantiated and are there to be extended. The `Internal` is not helping here (my opinion) but consfuses.

Reply:
I think with Internal and abstract it works as a building block and also consistent with other aggregations. Does adding java doc on top of the classes resolve the confusion?

@reta

Reply:
@navneet1v naming is always hard, `internal` in general is something that should not be used outside a very narrow scope (paralles to `@opensearch.internal`), which is not the case anymore here (we make it public). Hence the suggestion - if API is becoming public, let's name it so.

Reply:
Yeah thats a valid point. Thanks for providing this idea. +1 on this.

Reply:
@reta In other classes, i see either BaseInterfaceName or AbstractInterfaceName. i couldn't find any abstract class that ends with "Base". I don't have strong opinion on one over other, do you have? However, i slightly prefer Base as prefix over suffix :)

Reply:
@VijayanB it is up to us :)

Reply:
@reta I updated name with prefix. Can you take another look? Thanks

Reply:
@VijayanB my apologies, missed the notification